### ISSUE TITLE
python 2: client doesn't able to publish binary data

### ISSUE THREAD

**Issue by yoch** [eclipse/paho.mqtt.python#78]
This snippet of code cause a crash on **python 2** : 

``` python
import paho.mqtt.client as mqtt

p = mqtt.Client()
p.connect("iot.eclipse.org")
p.loop_start()

data = str(bytearray(range(256)))   # binary data
p.publish('a/b/c', data)
```

```
Traceback (most recent call last):
  File "C:/Users/yoch/Desktop/paho.mqtt/src/test-binary.py", line 8, in <module>
    p.publish('a/b/c', data)
  File "C:/Users/yoch/Desktop/paho.mqtt/src\paho\mqtt\client.py", line 974, in publish
    rc = self._send_publish(local_mid, topic, local_payload, qos, retain, False, info)
  File "C:/Users/yoch/Desktop/paho.mqtt/src\paho\mqtt\client.py", line 1979, in _send_publish
    upayload = payload.encode('utf-8')
UnicodeDecodeError: 'ascii' codec can't decode byte 0x80 in position 128: ordinal not in range(128)
```

The reason is that the client try incorrectly to encode data to utf8, while it'sbinary data.

**Comment by yoch** [comment]
Duplicate of issue #15, and properly fixed by PR #81.

**Comment by PierreF** [comment]
Indeed, it's fixed on develop branch.
